src/components/DropDetails.js

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react';
+import { ReactComponent as RowDown } from '../assets/images/chevron-up.svg';
+
+/**
+ * DropDetails component renders a collapsible section with a title and content body.
+ * The body section can be toggled open or closed by clicking on the title.
+ * Optionally, an external callback function can be triggered when the toggle occurs.
+ *
+ * @param {string} title - The title displayed in the header of the dropdown section.
+ * @param {boolean} startOpen - Determines if the dropdown is open by default when the component is first rendered.
+ * @param {Function} onT - Optional callback function triggered when the dropdown is toggled.
+ *                          This function is called every time the user clicks to open or close the dropdown.
+ * @param {ReactNode} children - The content displayed inside the dropdown body when it's open.
+ *
+ * @returns {JSX.Element} The DropDetails component, which includes a title and a collapsible body.
+ */
+export const DropDetails = ({ title, startOpen, onT, children }) => {
+  const [open, setOpen] = useState(startOpen);
+
+  /**
+   * Toggles the open/close state of the dropdown and calls the optional onT callback.
+   */
+  const click = () => {
+    if (onT) {
```

Comment:
suggestion(non-blocking): I'd recommend changing this parameter name to `onToggle` for more clarity.

Reply:
Thanks for the suggestion. Renamed parameter. Please review.🙂